src/skin/legacyskinparser.cpp

```diff
@@ -1084,6 +1089,29 @@ QWidget* LegacySkinParser::parseCoverArt(QDomElement node) {
     return pCoverArt;
 }
 
+void LegacySkinParser::parseSingletonDefinition(QDomElement node) {
+    // The actual singleton definition is identical to any other WWidgetGroup.
+    QWidget* child_group = parseWidgetGroup(node);
+    WSingletonContainer::defineSingleton(child_group->objectName(),
```

Comment:
I don't think it's leaked, ownership of the pointer is passed to the SingletonContainer class.  But I do have to figure out destruction -- if a user loads another skin, and then reloads the first skin, bad things may happen.